commit ac2f43d3d34e52b0d388b4c573ff6bbac90235b9
Author: Justin Stitt <justinstitt@google.com>
Date:   Wed Oct 25 20:47:56 2023 +0300

    wifi: ath10k: replace deprecated strncpy with memcpy
    
    strncpy() is deprecated [1] and we should prefer less ambiguous
    interfaces.
    
    In this case, arvif->u.ap.ssid has its length maintained by
    arvif->u.ap.ssid_len which indicates it may not need to be
    NUL-terminated. Make this explicit with __nonstring and use a plain old
    memcpy.
    
    This is also consistent with future copies into arvif->u.ap.ssid:
    
            if (changed & BSS_CHANGED_SSID &&
                vif->type == NL80211_IFTYPE_AP) {
                    arvif->u.ap.ssid_len = vif->cfg.ssid_len;
                    if (vif->cfg.ssid_len)
                            memcpy(arvif->u.ap.ssid, vif->cfg.ssid,
                                   vif->cfg.ssid_len);
                    arvif->u.ap.hidden_ssid = info->hidden_ssid;
            }
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Acked-by: Jeff Johnson <quic_jjohnson@quicinc.com>
    Signed-off-by: Kalle Valo <quic_kvalo@quicinc.com>
    Link: https://lore.kernel.org/r/20231024-strncpy-drivers-net-wireless-ath-ath10k-mac-c-v2-1-4c1f4cd4b4df@google.com

diff --git a/drivers/net/wireless/ath/ath10k/core.h b/drivers/net/wireless/ath/ath10k/core.h
index 4b5239de4018..ba9795a8378a 100644
--- a/drivers/net/wireless/ath/ath10k/core.h
+++ b/drivers/net/wireless/ath/ath10k/core.h
@@ -607,7 +607,7 @@ struct ath10k_vif {
 			u8 tim_bitmap[64];
 			u8 tim_len;
 			u32 ssid_len;
-			u8 ssid[IEEE80211_MAX_SSID_LEN];
+			u8 ssid[IEEE80211_MAX_SSID_LEN] __nonstring;
 			bool hidden_ssid;
 			/* P2P_IE with NoA attribute for P2P_GO case */
 			u32 noa_len;